### ISSUE TITLE
[FEATURE] Add a keybind to toggle show major cooldowns in separate cd's display or primary display

### ISSUE THREAD

**Issue by TheRealUncleFrank** [Hekili/hekili#3962]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/enhancement) for this request and was not able to find one.
- [X] I edited the title of this feature request (above) so that it describes the issue I am reporting.

### Feature Request

Under Toggles > Damage Cooldowns, there's a keybind and checkbox to "Enable major cooldowns", but only a checkbox to "Show in separate cd's display". I'd like a keybind for "Show in separate cd's display".
I always want major cd's enabled, but sometimes I want them in a separate display and sometimes I want them in the primary display, so a keybind to switch between the 2 would be great. I only use the major cd's on bigger fights/boss fights, and would like them in the primary display for that, but I don't want to disable them otherwise, just move them over to their own separate cd display, so there are still recs for them just in case.

**Comment by TreeOfLearning** [comment]
This would be really useful for defensives too!

**Comment by syrifgit** [comment]
Not sure if adding a whole keybinding option for it is necessary, but it could possibly be added as a macro command like `/hek set cooldowns separate`

**Comment by Hekili** [comment]
This is something I'd like to do, but I worry a bit about how to communicate the behavior. I'm not sure if it should work like the Mode toggle, where the same button would rotate through CDs OFF > ON > ON+Separate. Or if it should be its own option. If it's promoted to a toggle, I need a plan for the minimap menu, datatext, and other messaging to make it comprehensible at a glance.

**Comment by johnnylam88** [comment]
I'm not sure how this would function within how Cooldowns are currently treated. The option to show Cooldowns in a separate display also causes the primary recommendation window to consider these major CDs to be on cooldown unless you also check the option to use the actual cooldown for those abilities.

@TheRealUncleFrank Could you elaborate on your specific use-case on why you are wanting this toggle?

I think given Hekili's engine, it would be better to just have different displays for the information, much like Single-Target vs. AOE displays. We could have a display show recommendations with major cooldowns and another that shows them without the major cooldowns.

**Comment by TheRealUncleFrank** [comment]
I'd like the ability to quickly change "Show in separate cd's display" between checked and unchecked, whether it be a keybind or a macro command like @syrifgit suggested. Not sure if having 2 primary displays at the same time, one with major cd's and one without, would be good, unless we could cycle thru them like @Hekili mentioned.
The option is already there, "Show in separate cd's display". Just want an easier/faster way to check and uncheck it, without having to open options.

I like to have major cd's in the primary display for bigger fights/elite rares/boss fights, but otherwise have them in the separate cd display for smaller fights, trash mobs, open world stuff because I rarely use major cd's for that.

**Comment by johnnylam88** [comment]
I guess my question is why you would want that instead of just toggling showing your cooldowns completely. 

If you separate CDs into its own display, then the primary recommendations assume that your major CDs are on cooldown and will not properly suggest the proper resource pooling or target debuff application or whatever other setup is usually needed before popping your major CDs.

If you just toggle major CDs on and off completely, then when it's toggled on, the priorities will make suggestions to get you towards optimal CD usage.

**Comment by Hekili** [comment]
Yeah, I question the value of the feature, especially in the face of confusing users about it.

> If you separate CDs into its own display, then the primary recommendations assume that your major CDs are on cooldown and will not properly suggest the proper resource pooling or target debuff application or whatever other setup is usually needed before popping your major CDs.

*Technically*, there is a setting that turns off the forced CD behavior, but that also causes the priority to break down, typically, because any logic that prepares to pop CDs will repeat endlessly until the CDs actually get used.

It's shown in the Abilities section as `(GLOBAL) When Cooldowns Shown Separately, Use Actual Cooldown`.

**Comment by johnnylam88** [comment]
> *Technically*, there is a setting that turns off the forced CD behavior, but that also causes the priority to break down, typically, because any logic that prepares to pop CDs will repeat endlessly until the CDs actually get used.
> 
> It's shown in the Abilities section as `(GLOBAL) When Cooldowns Shown Separately, Use Actual Cooldown`.

Yes, I've used that setting at one point but I gave up on it because if I was going to use the actual cooldown, then I might as well just show them in the normal recommendation queue so I can more easily see exactly when the cooldown should be used.

I believe that the best way to use Hekili is to not try to split out abilities into separate displays.

**Comment by TheRealUncleFrank** [comment]
The option to show the CDs in a separate display is already there, and I assume being used by many more people than just me, since the option is so easy to see, so this isn't something completely new.
I just want a quicker/easier way to check/uncheck that box, even if it's just a macro like @syrifgit suggested, if a keybind is too complex to add.

> I think given Hekili's engine, it would be better to just have different displays for the information, much like Single-Target vs. AOE displays. We could have a display show recommendations with major cooldowns and another that shows them without the major cooldowns.

Which is basically pretty much what I want. I think I was just trying to phrase it in the framework of the options that are currently availble. You suggest 2 primary displays, one with and one without major cds. My original idea was to just toggle the single primary display back and forth between those 2 options, which is what I assume that checkbox option kinda does by removing the cd recs from the primary display. But now that you've mentioned it, I think I might like 2 separate primary displays better, one with major cds recommendations and one without.

**Comment by syrifgit** [comment]
![image](https://github.com/user-attachments/assets/67f45d83-21b4-4b6b-a0e6-45379cbdb235)


Currently implemented: Chat commands for control of toggle box no. 1 from the screenshot
`/hek set cooldowns` - toggle cooldowns
`/hek set cooldowns [on/off]` - set the cooldowns toggle to the provided value


Proposal: Add these chat commands to control toggle box no. 2 from the screenshot
`/hek set cooldowns separate` - toggle the cooldowns separate checkbox, has no bearing on the overall enable/disable state of cooldowns
`/hek set cooldowns separate [on/off]` - toggle the cooldowns separate checkbox, has no bearing on the overall enable/disable state of cooldowns

**Comment by TheRealUncleFrank** [comment]
Excellent!
Either those chat commands, or 2 separate primary displays like @johnnylam88 suggested, one with major cd recs and one without.

**Comment by syrifgit** [comment]
This should be getting merged soon. Here is the new reference sheet for macros: <https://github.com/Hekili/hekili/wiki/Hekili,-Macros-and-You>

> Proposal: Add these chat commands to control toggle box no. 2 from the screenshot
/hek set cooldowns separate - toggle the cooldowns separate checkbox, has no bearing on the overall enable/disable state of cooldowns
/hek set cooldowns separate [on/off] - toggle the cooldowns separate checkbox, has no bearing on the overall enable/disable state of cooldowns

This example I gave before will work.
